File: 이승민/[20.04.07]1697.py
from collections import deque

def bfs():
    while q:
        num = q.popleft()
        if num == k: return visited[num]

        if 0 <= (num - 1) and not visited[num - 1]:
            visited[num - 1] = visited[num] + 1
            q.append(num-1)

        if num + 1 <= 100000 and not visited[num + 1]:
            visited[num + 1] = visited[num] + 1
            q.append(num+1)

        if num * 2 <= 100000 and not visited[num * 2]:
            visited[num * 2] = visited[num] + 1
            q.append(num*2)

if __name__ == "__main__":
    n, k = map(int, input().split())
    visited = [0 for i in range(100001)]
    q = deque()
    q.append(n)
    visited[n] = 0
    print(bfs())

File: 이승민/[20.04.07]1697.cpp
#include <iostream>
#include <queue>

using namespace std;

int n = 0, k = 0, visited[100001] = {0};
queue<int> q;

int bfs(){

    while( !q.empty() ){

        int num = q.front();
        q.pop();

        if( num == k ) return visited[num];

        if( 0 <= (num - 1) && !visited[num - 1] ){
            visited[num - 1] = visited[num] + 1;
            q.push(num-1);
        }

        if( num + 1 <= 100000 && !visited[num + 1] ) {
            visited[num + 1] = visited[num] + 1;
            q.push(num+1);
        }

        if( num * 2 <= 100000 && !visited[num * 2] ) {
            visited[num * 2] = visited[num] + 1;
            q.push(num*2);
        }
    }
}

int main(){
    cin >> n >> k;
    q.push(n);
    printf("%d", bfs());

    return 0;
}
